Test/Unit/Preprocessor/Adapter/Scss/ProcessorTest.php

```diff
@@ -26,99 +29,180 @@ class ProcessorTest extends \PHPUnit_Framework_TestCase
 
     const TEST_EXCEPTION_MESSAGES = 'Test exception messages';
 
+    private $compiler;
+
+    /**
+     * Setup before each test
+     */
+    public function setUp()
+    {
+        $this->compiler = new Compiler();
+    }
+
     /**
-     * Run test for processContent method
+     * SCSS should be processed successfully under normal conditions
      */
     public function testProcessContent()
     {
         $fileMock = $this->getFileMock();
-        $loggerMock = $this->getLoggerMock();
-        $assetSourceMock = $this->getSourceMock();
 
-        $loggerMock->expects(self::never())
-            ->method('critical');
+        $assetSourceMock = $this->getSourceMock();
+        $loggerMock = $this->getLoggerMock();
+        $stateMock = $this->getStateMock();
 
         $assetSourceMock->expects(self::once())
             ->method('getContent')
             ->with($fileMock)
             ->willReturn(file_get_contents(__DIR__ . self::TEST_FILE)); // @codingStandardsIgnoreLine
 
-        $processor = new Processor($assetSourceMock, $loggerMock);
+        $loggerMock->expects(self::never())
+            ->method('critical');
+
+        $processor = new Processor($assetSourceMock, $loggerMock, $stateMock, $this->compiler);
 
         $content = $processor->processContent($fileMock);
 
         $search = [' ', "\t", "\n", "\r", "\0", "\x0B"];
-        $expectedContent = 'h1{color:#009a82;margin:0;padding:0;}#container{width:460px;margin:0pxauto;}';
+        $expectedContent = <<<'EOT'
+/*line12*/h1{color:#009a82;margin:0;padding:0;}/*line18*/#container{width:460px;margin:0pxauto;}
```

Comment:
As will said in slack, `margin:0pxauto;` is invalid CSS. Is that what the preprocessor implementation is spitting out?

Reply:
The test was comparing output after removing characters specified in the `$search` list.